eth/downloader/downloader.go

```diff
@@ -587,41 +587,87 @@ func (d *Downloader) fetchHeight(p *peerConnection) (*types.Header, error) {
 	}
 }
 
+// calcRequestSpan calculates what headers to request from a peer when trying to determine the
+// common ancestor.
+func calculateRequestSpan(remoteHeight, localHeight uint64) (int64, int, int, uint64) {
+	var (
+		from     int
+		count    int
+		MaxCount = MaxHeaderFetch / 16
+	)
+	// requestHead is the highest block that we will ask for. If requestHead is not offset,
+	// the highest block that we will get is 16 blocks back from head, which means we
+	// will fetch 14 or 15 blocks unneccesarily in the case the height difference
+	// between us and the peer is 1-2 blocks, which is most common
+	requestHead := int(remoteHeight) - 1
+	if requestHead < 0 {
+		requestHead = 0
+	}
+	// requestBottom is the lowest block we want included in the query
+	// Ideally, we want to include just below own head
+	requestBottom := int(localHeight - 1)
+	if requestBottom < 0 {
+		requestBottom = 0
+	}
+	totalSpan := requestHead - requestBottom
+	span := 1 + totalSpan/MaxCount
+	if span < 2 {
+		span = 2
```

Comment:
An particular reason for putting the minimum span to 2? Maybe if we're close enough we can request head-1 and head-2 instea of head-3? Not sure, thinking out aloud here really.

Reply:
`span` is not the same as `skip`. So `span` of `2` means we're getting every `2`nd item, so `skip` is `1`.